### PR TITLE
Create new instance for fee calculation

### PR DESCRIPTION
**Pull request by jusbar23**
### Changelist
* Fix issue with fee calculation for ORRS

### Test Plan
* Test cases

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

* **Bug Fixes**
  * Prevented unintended mutation of net fees during revenue-share calculations by using mutation‑safe copies and adjusting subtraction logic; added a post‑aggregation validation that logs and aborts if total shared fees exceed net fees.

* **Tests**
  * Added comprehensive tests covering taker/maker order‑router rev shares, unconditional rev shares, maker rebates, and affiliate‑whitelist interactions to validate aggregation, allocation, and edge cases.

### CODE REVIEW

File: protocol/x/revshare/keeper/revshare.go [hunk 1]
```diff
@@ -220,7 +220,7 @@ func (k Keeper) GetAllRevShares(
 	}
 
 	var orderRouterRevShares []types.RevShare
-	netFeesSubRevenueShare := netFees
+	netFeesSubRevenueShare := new(big.Int).Set(netFees)
```
Comment (anmolagrawal345) [inline protocol/x/revshare/keeper/revshare.go:223]: nit: line 233 in the else - lets make it netFeesSubRevenueShare.Sub(netFeesSubRevenueShare, affiliateFeesShared)?

Comment (jusbar23) [reply protocol/x/revshare/keeper/revshare.go:223]: Done

**Comment by jusbar23** [comment]
@mergifyio backport release/indexer/v9.x

**Comment by jusbar23** [comment]
@mergifyio backport release/protocol/v9.x
